File: lab7/secondTask/MainActivity.java
package com.example.myapplicationrec;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    BroadcastReceiver broadcastReceiver;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        broadcastReceiver = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                String msg = intent.getStringExtra("PATH");
                TextView textView = findViewById(R.id.textView);
                textView.setText(msg);
            }
        };
        registerReceiver(broadcastReceiver, new IntentFilter("BROADCAST"));

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(broadcastReceiver);
    }

}

File: Лабораторные/AndroidLab6/src - Threads/main/java/ru/spbstu/icc/kspt/lab2/continuewatch/MainActivity.kt
package ru.spbstu.icc.kspt.lab2.continuewatch

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*
import java.util.concurrent.locks.*;

class MainActivity : AppCompatActivity() {
    private var secondsElapsed: Int = 0
    private var background: Thread? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        background = Thread {
            try {
                var timePast: Long
                var delay = 1000L
                var delayPast = 1000L
                var time = System.currentTimeMillis() - 1000
                while (true) {
                    timePast = time
                    time = System.currentTimeMillis()
                    delay -= time - timePast - delayPast
                    Log.d("Timer", delay.toString())
                    Thread.sleep(delay)
                    textSecondsElapsed.post {
                        textSecondsElapsed.setText("Seconds elapsed: " + secondsElapsed++)
                    }
                    if (background?.isInterrupted == true) break;
                    delayPast = delay
                    delay = 1000L
                }
            } catch (e: InterruptedException) {
            }
        }
        background?.start()
        super.onResume()
    }

    override fun onPause() {
        background?.interrupt()
        background = null
        super.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt("secs", secondsElapsed)
        super.onSaveInstanceState(outState)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        secondsElapsed = savedInstanceState.getInt("secs")
        textSecondsElapsed.setText("Seconds elapsed: " + secondsElapsed++)

        super.onRestoreInstanceState(savedInstanceState)
    }
}
